cmd/proxy/actions/cache_miss_handler.go

```diff
@@ -48,7 +46,9 @@ func isModuleNotFoundErr(err error) bool {
 	if _, ok := err.(storage.ErrVersionNotFound); ok {
 		return ok
 	}
-
-	s := err.Error()
-	return strings.HasPrefix(s, "module ") && strings.HasSuffix(s, "not found")
+	if err != nil {
+		s := err.Error()
+		return strings.HasPrefix(s, "module ") && strings.HasSuffix(s, "not found")
```

Comment:
any way to get away with checking error types instead of string matching?

Reply:
i think we can, it was not working for me, because FS did not return type error and mongo did. so i fixed FS and one occurrence of string match, this slipped through